Annotate Media Stream endpoints with MediaDevicesEnabled

darryl-apple opened the pull request:
#### 63c515a99f14ff20cea263edc954ca34fbf9cb29
<pre>
Annotate Media Stream endpoints with MediaDevicesEnabled
<a href="https://bugs.webkit.org/show_bug.cgi?id=296429">https://bugs.webkit.org/show_bug.cgi?id=296429</a>
<a href="https://rdar.apple.com/156599811">rdar://156599811</a>

Reviewed by NOBODY (OOPS!).

Guard Media Stream and Capture endpoints with correct runtime flags

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Modules/mediastream/MediaDevices.idl:
* Source/WebKit/GPUProcess/webrtc/RemoteAudioMediaStreamTrackRendererInternalUnitManager.messages.in:
* Source/WebKit/GPUProcess/webrtc/UserMediaCaptureManagerProxy.h:
* Source/WebKit/GPUProcess/webrtc/UserMediaCaptureManagerProxy.messages.in:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
(WebKit::NetworkConnectionToWebProcess::updateSharedPreferencesForWebProcess):
* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::rtcDataChannelProxy):
* Source/WebKit/NetworkProcess/webrtc/NetworkMDNSRegister.cpp:
(WebKit::NetworkMDNSRegister::sharedPreferencesForWebProcess const):
* Source/WebKit/NetworkProcess/webrtc/NetworkMDNSRegister.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkMDNSRegister.messages.in:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCMonitor.cpp:
(WebKit::NetworkRTCMonitor::sharedPreferencesForWebProcess const):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCMonitor.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCMonitor.messages.in:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.cpp:
(WebKit::NetworkRTCProvider::NetworkRTCProvider):
(WebKit::NetworkRTCProvider::sharedPreferencesForWebProcess):
(WebKit::NetworkRTCProvider::updateSharedPreferencesForWebProcess):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.messages.in:
* Source/WebKit/NetworkProcess/webrtc/RTCDataChannelRemoteManagerProxy.cpp:
(WebKit::RTCDataChannelRemoteManagerProxy::RTCDataChannelRemoteManagerProxy):
(WebKit::RTCDataChannelRemoteManagerProxy::registerConnectionToWebProcess):
(WebKit::RTCDataChannelRemoteManagerProxy::unregisterConnectionToWebProcess):
(WebKit::RTCDataChannelRemoteManagerProxy::sharedPreferencesForWebProcess):
* Source/WebKit/NetworkProcess/webrtc/RTCDataChannelRemoteManagerProxy.h:
* Source/WebKit/NetworkProcess/webrtc/RTCDataChannelRemoteManagerProxy.messages.in:
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::anyPageHasMediaDevicesEnabled const):
(WebKit::WebProcess::anyPageHasSpeechRecognitionEnabled const):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/cocoa/RemoteRealtimeAudioSource.cpp:
(WebKit::RemoteRealtimeAudioSource::setIsInBackground):
* Source/WebKit/WebProcess/cocoa/UserMediaCaptureManager.cpp:
(WebKit::UserMediaCaptureManager::setupCaptureProcesses):
(WebKit::UserMediaCaptureManager::AudioFactory::createAudioCaptureSource):
(WebKit::UserMediaCaptureManager::VideoFactory::createVideoCaptureSource):
(WebKit::UserMediaCaptureManager::DisplayFactory::createDisplayCaptureSource):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/63c515a99f14ff20cea263edc954ca34fbf9cb29

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/114061 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/33813 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/24271 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/120230 "Built successfully") | [❌ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/65054 "Failed to checkout and rebase branch from PR 48476") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/115950 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/34441 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/42373 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86693 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/41691 "Unable to confirm if test failures are introduced by change, retrying build") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/117009 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/27423 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/102443 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67101 "Found 1 new API test failure: /WPE/TestCookieManager:/webkit/WebKitCookieManager/persistent-storage (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26608 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20571 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/63934 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/106489 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96790 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20688 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/123457 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/112624 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/41093 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30616 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95525 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/41471 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98652 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/95308 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40440 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/18248 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/37251 "Failed to checkout and rebase branch from PR 48476") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/40969 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/46471 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/136825 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/40594 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/36635 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/43894 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/42348 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/51c3177ff8f2601ad41e705c3eba5bb3d93edc96. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48476

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ccee46d83aa095e958a39f70f416c8a547eaf157. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48476

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/aa328755cdfc10cb4d46c50e2194d691f169442c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48476

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3b54854d85eda40babfd6d21eca387df8e07fc06. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48476

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2be4c410102cc7d301fae92d3136b573d15084a7. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48476

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3f02533b07d022ec18935b4c80065cb770768c66. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48476

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/29522ebb8b0c487d32ddbcb30635389e1d61ba60. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48476

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/cb9e812ddfa07d0e7afce3473519c0a71d9e9e46. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48476

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c7c04f94dff7ce7e696dac88f3dda733aab1b7b9. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48476

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c7c04f94dff7ce7e696dac88f3dda733aab1b7b9. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48476

darryl-apple committed:
Annotate Media Stream endpoints with MediaDevicesEnabled
https://bugs.webkit.org/show_bug.cgi?id=296429
rdar://156599811

Reviewed by NOBODY (OOPS!).

Guard Media Stream and Capture endpoints with correct runtime flags

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Modules/mediastream/MediaDevices.idl:
* Source/WebKit/GPUProcess/webrtc/RemoteAudioMediaStreamTrackRendererInternalUnitManager.messages.in:
* Source/WebKit/GPUProcess/webrtc/UserMediaCaptureManagerProxy.h:
* Source/WebKit/GPUProcess/webrtc/UserMediaCaptureManagerProxy.messages.in:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
(WebKit::NetworkConnectionToWebProcess::updateSharedPreferencesForWebProcess):
* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::rtcDataChannelProxy):
* Source/WebKit/NetworkProcess/webrtc/NetworkMDNSRegister.cpp:
(WebKit::NetworkMDNSRegister::sharedPreferencesForWebProcess const):
* Source/WebKit/NetworkProcess/webrtc/NetworkMDNSRegister.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkMDNSRegister.messages.in:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCMonitor.cpp:
(WebKit::NetworkRTCMonitor::sharedPreferencesForWebProcess const):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCMonitor.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCMonitor.messages.in:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.cpp:
(WebKit::NetworkRTCProvider::NetworkRTCProvider):
(WebKit::NetworkRTCProvider::sharedPreferencesForWebProcess):
(WebKit::NetworkRTCProvider::updateSharedPreferencesForWebProcess):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.messages.in:
* Source/WebKit/NetworkProcess/webrtc/RTCDataChannelRemoteManagerProxy.cpp:
(WebKit::RTCDataChannelRemoteManagerProxy::RTCDataChannelRemoteManagerProxy):
(WebKit::RTCDataChannelRemoteManagerProxy::registerConnectionToWebProcess):
(WebKit::RTCDataChannelRemoteManagerProxy::unregisterConnectionToWebProcess):
(WebKit::RTCDataChannelRemoteManagerProxy::sharedPreferencesForWebProcess):
* Source/WebKit/NetworkProcess/webrtc/RTCDataChannelRemoteManagerProxy.h:
* Source/WebKit/NetworkProcess/webrtc/RTCDataChannelRemoteManagerProxy.messages.in:
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::anyPageHasMediaDevicesEnabled const):
(WebKit::WebProcess::anyPageHasSpeechRecognitionEnabled const):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/cocoa/RemoteRealtimeAudioSource.cpp:
(WebKit::RemoteRealtimeAudioSource::setIsInBackground):
* Source/WebKit/WebProcess/cocoa/UserMediaCaptureManager.cpp:
(WebKit::UserMediaCaptureManager::setupCaptureProcesses):
(WebKit::UserMediaCaptureManager::AudioFactory::createAudioCaptureSource):
(WebKit::UserMediaCaptureManager::VideoFactory::createVideoCaptureSource):
(WebKit::UserMediaCaptureManager::DisplayFactory::createDisplayCaptureSource):

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/110fd8003dcd8980399808ba6d424759ca72a424. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48476

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/63c515a99f14ff20cea263edc954ca34fbf9cb29. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48476
